gravitee-am-service/src/main/java/io/gravitee/am/service/impl/I18nDictionaryService.java

```diff
@@ -0,0 +1,254 @@
+                                throw new DictionaryAlreadyExistsException(updateDictionary.getName());
+                            }
+                            return existingDictionary;
+                        })
+                )
+                .flatMapSingle(oldDictionary -> {
+                    var toUpdate = new I18nDictionary(oldDictionary);
+                    if (updateDictionary.getName() != null) {
+                        toUpdate.setName(updateDictionary.getName());
+                    }
+                    if (updateDictionary.getLocale() != null) {
+                        toUpdate.setLocale(updateDictionary.getLocale());
+                    }
+                    if (updateDictionary.isDefaultLocale() != null) {
+                        toUpdate.setDefaultLocale(updateDictionary.isDefaultLocale());
+                    }
+                    if (updateDictionary.getEntries() != null) {
+                        toUpdate.setEntries(updateDictionary.getEntries());
+                    }
+                    return repository.update(toUpdate);
+                })
+                .flatMap(dictionary -> {
+                    Event event = new Event(Type.I18N_DICTIONARY, new Payload(dictionary.getId(), dictionary.getReferenceType(), dictionary.getReferenceId(), UPDATE));
+                    return eventService
+                            .create(event)
+                            .flatMap(createdEvent -> Single.just(dictionary));
+                })
+                .onErrorResumeNext(ex -> {
+                    if (ex instanceof AbstractManagementException) {
+                        return Single.error(ex);
+                    }
+                    String msg = "An error occurs while trying to update a dictionary";
+                    LOGGER.error(msg, ex);
+                    return Single.error(new TechnicalManagementException(msg, ex));
+                });
+    }
+
+    public Single<I18nDictionary> updateEntries(ReferenceType referenceType, String referenceId, String id, Map<String, String> entries) {
+        LOGGER.debug("Update entries for dictionary {} for {} {}", id, referenceType, referenceId);
+        return findById(referenceType, referenceId, id)
+                .flatMapSingle(oldDictionary -> {
+                    var toUpdate = new I18nDictionary(oldDictionary);
+                    toUpdate.setEntries(entries);
+                    return repository.update(toUpdate);
+                })
+                .flatMap(dictionary -> {
+                    Event event = new Event(Type.I18N_DICTIONARY, new Payload(dictionary.getId(), dictionary.getReferenceType(), dictionary.getReferenceId(), UPDATE));
+                    return eventService
+                            .create(event)
+                            .flatMap(createdEvent -> Single.just(dictionary));
+                })
+                .onErrorResumeNext(ex -> {
+                    if (ex instanceof AbstractManagementException) {
```

Comment:
As we create an audit entry for the create action, we should also do it for the Update